src/Parsers/ParserDataType.cpp

```diff
@@ -57,6 +58,13 @@ bool ParserDataType::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
         return false;
     tryGetIdentifierNameInto(identifier, type_name);
 
+    /// Don't accept things like Array(`x.y`).
```

Comment:
That's strange.
What about `Enum('Hello~@#', ' w\0 r \td\\')`?

Reply:
Still works (checked). That's literals, this only applies to identifiers.